container-storage-setup.sh

```diff
@@ -431,6 +431,50 @@ check_docker_storage_metadata() {
   exit 1
 }
 
+# See whether we can use overlay backend
+can_use_overlay() {
+    local mountroot=$1
+    fstype=$(stat -f -c '%T' $mountroot)
+    have_ftype=yes
+    case "$fstype" in
+        # For xfs, see https://bugzilla.redhat.com/show_bug.cgi?id=1288162#c8
+        xfs) if test "$(xfs_info $mountroot | grep -o 'ftype=[01]')" = "ftype=0"; then
+                 have_ftype=no
+                 Info "XFS filesystem at ${mountroot} has ftype=0, cannot use overlay backend; consider separate volume (or OS reprovision)"
+             fi
+             ;;
+       *)
+             # Punt and assume things are OK
+             ;;
+    esac
+    if test ${have_ftype} = yes; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# If STORAGE_DRIVER=auto, pick a backend, and update STORAGE_DRIVER
+select_auto_storage_driver() {
+    local mountroot=$1
+    if [ "$STORAGE_DRIVER" == "auto" ]; then
+        # Is a CONTAINER_THINPOOL option set to a non-default value, or
+        # the "VG" option is set?
+        # Then choose "devicemapper".
+        if [ "${CONTAINER_THINPOOL:-}" != "${_DEFAULT_CONTAINER_THINPOOL}" ] ||
```

Comment:
Oh using here.

Hmm.., so what happens if user specifies `CONTAINER_THINPOOL=docker-pool` in /etc/sysconfig/docker-storaege-setup. 

So how about this. 

Lets not change our STORAGE_DRIVER default and let that be devicemapper. And ask user to specify STORAGE_DRIVER=auto.

That way distributions can ship their default config file with STORAGE_DRIVER=auto. And this will take affect only on new installations. 

And in simplest form, we will ignore all driver specific options (instead of guessing like above). Down the line we could harden the code to error out.